windows/src/engine/kmtip/README.md

```diff
@@ -0,0 +1,102 @@
+
+## Overview
+
+`kmtip` is an implementation of a Text Service for the Text Services Framework
+(TSF). This implementation is named Keyman TIP. Architecturally it sits between
+the TSF Manager and Keyman.exe.
+
+## Background
+
+### 1. The "Mixed Architecture" Environment
+
+Windows on both amd64(x86-64) and on Arm devices can run applications of differing architectures simultaneously:
+
+On amd64 Devices:
+* **Native x64 apps:** (e.g., Windows Explorer, Notepad, Edge).
+* **x86 apps (WOW64):** (32-bit legacy apps).
+
+On Arm Devices:
+* **Native Arm64 apps:** (e.g., Windows Explorer, Notepad, Edge).
+* **Emulated x64 apps:** (e.g., Older desktop applications).
+* **Emulated x86 apps:** (32-bit legacy apps).
+
+A Text Input Processor (TIP) is a plugin that gets **loaded directly into the
+process space** of the application currently receiving text input. Therefore,
+the TIP DLL must match the architecture of the application it is running in.
+Setting all this up is different on Windows on Arm and Windows on amd64
+devices.
+
+
+## kmtip.dll and kmtip64.dll - On amd64 Devices
+
+On 64-bit Windows, there is a single registry entry (COM CLSID) for a 64-bit
+TIP: **kmtip64.dll**.
+
+For 32-bit applications, there exists a WOW6432Node in the Windows registry
+that allows a different DLL to be registered for the same Class ID:
+**kmtip.dll**. It is essentially two views of the registry depending on whether
+the application is 32-bit or 64-bit.
+
+
+## kmtiparm64x.dll - Arm64 Devices
+
+### Single Registry Entry Problem
+
+On Windows on Arm there is a problem because there is only a single registry
+entry for the 64-bit CLSID. This means it is not possible to register both an
+amd64 DLL and an Arm64 DLL, for example **kmtip64.dll** and
+**kmtiparm64.dll**.
+
+* If we register a standard **Arm64** DLL, x64 apps will crash or fail to load
+  it because they cannot load a pure Arm64 binary.
+* If we register a standard **x64** DLL, native Arm64 apps will fail to load
+  it.
+
+### The Solution: Arm64X
+
+**Arm64X** is a PE (Portable Executable) binary format. It is a dual-architecture
+library
+capable of servicing both native Arm64 applications and emulated x64
+applications on Windows on Arm (WoA) devices.
+
+`kmtiparm64x.dll` solves this problem by using the **Arm64X** format. It
+contains two complete sets of code within a single file:
+
+1. **Arm64 Code:** Used when the DLL is loaded by a native Arm64 process.
+2. **Arm64EC (Emulation Compatible) Code:** Used when the DLL is loaded by an
```

Comment:
What about emulated 32bit apps that you mentioned above? Does that use Arm64EC code because the emulator deals with it?